app/src/main/java/com/shagalalab/constitution/list/ListFragment.kt

```diff
@@ -0,0 +1,44 @@
+package com.shagalalab.constitution.list
+
+import android.os.Bundle
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.recyclerview.widget.DividerItemDecoration
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.shagalalab.constitution.R
+import com.shagalalab.constitution.list.adapter.ItemClickListener
+import com.shagalalab.constitution.list.adapter.ListAdapter
+import com.shagalalab.constitution.list.model.Item
+import kotlinx.android.synthetic.main.fragment_list.*
+
+class ListFragment : AppCompatActivity(), ItemClickListener {
+
+    companion object {
+        const val k = 10
+    }
+
+    private var lang = ""
+    private val adapter = ListAdapter(this)
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.fragment_list)
+        lang = intent.getStringExtra("lang")
+        list.adapter = adapter
+        list.layoutManager = LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false)
+        list.addItemDecoration(DividerItemDecoration(this, DividerItemDecoration.VERTICAL))
+        adapter.setData(getData())
+    }
+
+    private fun getData(): MutableList<Item> {
```

Comment:
Do you know the difference between `List` and `MutableList`? Basically, `List` is immutable, that means once you create a list, you cannot change it. `MutableList` is as its name says mutable, you can change the list whatever way you'd like.

Here, you don't change the outcome of `getData()` method, you just pass it to adapter. That means, the signature of this method can be just `List`, and inside the method you can use `MutableList` if you wish. This will work, because `MutableList` is subclass (subinterface to be specific) of `List`.